### PR TITLE
Move Amazon & basic account classes to Sashimi

### PR DESCRIPTION
**Pull request by flin-8**

### CODE REVIEW

File: source/Server.Contracts/Accounts/AccountDetails.cs [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        {
+            return Enumerable.Empty<Variable>();
+        }
+
+        public virtual IEnumerable<(string key, string template)> ContributeResourceLinks()
+        {
+            return Enumerable.Empty<(string key, string template)>();
+        }
+
+        public abstract IEnumerable<Variable> ContributeVariables();
```
Comment (flin-8) [inline source/Server.Contracts/Accounts/AccountDetails.cs:23]: I made this abstract, I think `Account` on the server side should contribute the `AccountType` variable

File: source/Server.Contracts/Accounts/IAccountTypeProvider.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
+using System;
+using System.Collections.Generic;
+using FluentValidation;
+using Octopus.Server.Extensibility.Extensions.Mappings;
+
+namespace Sashimi.Server.Contracts.Accounts
+{
+    public interface IAccountTypeProvider: IContributeMappings, IServiceMessageHandler
```
Comment (flin-8) [inline source/Server.Contracts/Accounts/IAccountTypeProvider.cs:8]: Some of these signatures are different from Server, do the same refactoring on the Server side before deleting the Server version

File: source/Server.Contracts/Accounts/IExpandVariableForAccountDetails.cs [hunk 3]
```diff
@@ -0,0 +1,13 @@
+using System.Collections.Generic;
+using Sashimi.Server.Contracts.Variables;
+
+namespace Sashimi.Server.Contracts.Accounts
+{
+    public interface IExpandVariableForAccountDetails
```
Comment (flin-8) [inline source/Server.Contracts/Accounts/IExpandVariableForAccountDetails.cs:6]: Some of these signatures are different from Server, do the same refactoring on the Server side before deleting the Server version

File: source/Server.Contracts/Accounts/AccountDetails.cs (hunk 1 above)
Comment (johnsimons) [reply source/Server.Contracts/Accounts/AccountDetails.cs:23]: This used to be
```
public virtual IEnumerable<Variable> ContributeVariables()
{
    yield return new Variable("Octopus.Account.AccountType", AccountType.ToString());
}
```
With this shape we are putting that responsibility on all account implementers, I am not sure that is their responsibility?

Comment (johnsimons) [reply source/Server.Contracts/Accounts/AccountDetails.cs:23]: actually all good, that was wrong that contribution is already done on the account 👍

Comment (flin-8) [reply source/Server.Contracts/Accounts/AccountDetails.cs:23]: Amazon now needs an empty implementation, which I think is OK. This saves Azure from having to do:
```
                foreach (var variable in base.ContributeVariables())
                {
                    yield return variable;
                }
```

**Pull request merged**